backend/src/main/java/com/nikogrid/backend/services/AuthenticationService.java

```diff
@@ -0,0 +1,61 @@
+package com.nikogrid.backend.services;
+
+import com.nikogrid.backend.configurations.SecurityConfig;
+import com.nikogrid.backend.dto.AuthenticationDTO;
+import com.nikogrid.backend.dto.RecoveryJwtTokenDTO;
+import com.nikogrid.backend.dto.RegisterDTO;
+import com.nikogrid.backend.entities.Role;
+import com.nikogrid.backend.entities.User;
+import com.nikogrid.backend.entities.UserDetailsImpl;
+import com.nikogrid.backend.exceptions.DuplicateUserException;
+import com.nikogrid.backend.repositories.UserRepository;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class AuthenticationService {
+    private AuthenticationManager authenticationManager;
+    private JwtTokenService jwtTokenService;
+    private UserRepository userRepository;
+    private SecurityConfig securityConfiguration;
```

Comment:
You only seem to want `SecurityConfig` for the `PasswordEncoder`, so instead depend on `PasswordEncoder` directly.